packages/layout/src/IconList.ts

```diff
@@ -0,0 +1,86 @@
+import { Entity, HTMLWidget } from "@hpcc-js/common";
+import { Html } from "@hpcc-js/other";
+import { select as d3Select } from "d3-selection";
+import { HorizontalList } from "./HorizontalList";
+import { VerticalList } from "./VerticalList";
+
+import "../src/IconList.css";
+
+export class IconList extends HTMLWidget {
+    protected _list;
+    protected _entity_list = [];
+    protected _content_list = [];
+    constructor() {
+        super();
+        this._list = new HorizontalList()
+            .orientation_default("horizontal")
+            .flexWrap_default("nowrap")
+            ;
+    }
+    enter(d, e) {
+        super.enter(d, e);
+        e
+            .style("height", "100%")
+            .style("width", "100%")
+            ;
+        d3Select(d.parentElement)
+            .style("height", "100%")
+            .style("width", "100%")
+            ;
+        this._list.target(d);
+    }
+    update(d, e) {
+        super.update(d, e);
+        this.data().forEach((row, idx) => {
+            if (!this._list.widgets()[idx]) {
```

Comment:
tweak code so it doesn't keep calling `_list.widgets()` - just call it once at the start.